src/js/controller/model.js

```diff
@@ -161,20 +164,23 @@ define([
         };
 
         this.changeVideoProvider = function(Provider) {
-            var container;
 
             if (_provider) {
                 _provider.off(null, null, this);
-                container = _provider.getContainer();
-                if (container) {
+                if (_provider.getContainer()) {
                     _provider.remove();
                 }
             }
 
             _currentProvider = new Provider(_this.get('id'), _this.getConfiguration());
 
+            var container = this.get('mediaContainer');
             if (container) {
                 _currentProvider.setContainer(container);
+            } else {
+                this.once('change:mediaContainer', function(model, container) {
```

Comment:
These `once` listeners are dangerous. This one needs to be cleared when _currentProvider changes, incase that happens before the event is fired.